k8-operator/api/v1alpha1/infisicalsecret_types.go

```diff
@@ -147,6 +147,20 @@ type MangedKubeSecretConfig struct {
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:default:=Orphan
 	CreationPolicy string `json:"creationPolicy"`
+
+	// The template to transform the secret data
+	// +kubebuilder:validation:Optional
+	Template *InfisicalSecretTemplate `json:"template,omitempty"`
```

Comment:
Question: How come we are making it a pointer here? (*InfisicalSecretTemplate)

Reply:
If not given it's a nil.

Reply:
Yeah just seems like a new pattern? We aren't using pointers anywhere else